dch/mqtt-subscriber/mqtt-subscriber.js

```diff
@@ -96,16 +122,153 @@ dcappClient.on(MQTT_MESSAGE_EVENT, function (topic, message) {
+      batteryDeviceMap[watchId] = 1;
+    }
+    if (dataObj.battery > 0.3) {
+      batteryDeviceMap[watchId] = 1;
+    } else if (lastThreshold >= 0.01 && dataObj.battery < 0.01) {
+      batteryDeviceMap[watchId] = dataObj.battery;
+      createAlert({
+        timestamp: dataObj.timestamp,
+        watch_id: watchId,
+        alert_type: 'LOW_BATTERY',
+        alert_text: 'Battery life is under 1%.'
+      });
+    } else if (lastThreshold >= 0.02 && dataObj.battery < 0.02) {
+      batteryDeviceMap[watchId] = dataObj.battery;
+      createAlert({
+        timestamp: dataObj.timestamp,
+        watch_id: watchId,
+        alert_type: 'LOW_BATTERY',
+        alert_text: 'Battery life is under 2%.'
+      });
+    } else if (lastThreshold >= 0.05 && dataObj.battery < 0.05) {
+      batteryDeviceMap[watchId] = dataObj.battery;
+      createAlert({
+        timestamp: dataObj.timestamp,
+        watch_id: watchId,
+        alert_type: 'LOW_BATTERY',
+        alert_text: 'Battery life is under 5%.'
+      });
+    } else if (lastThreshold >= 0.1 && dataObj.battery < 0.1) {
+      batteryDeviceMap[watchId] = dataObj.battery;
+      createAlert({
+        timestamp: dataObj.timestamp,
+        watch_id: watchId,
+        alert_type: 'LOW_BATTERY',
+        alert_text: 'Battery life is under 10%.'
+      });
+    }
+
+    // Reset timeout delay if it is in the connectedDevicesMap. Otherwise,
+    // append an alert that a new device has connected and insert it into the
+    // database
+    if (connectedDeviceMap[watchId] !== undefined) {
+      console.log(connectedDeviceMap[watchId].toString());
+      var timeout = connectedDeviceMap[watchId];
+      clearTimeout(timeout);
+      connectedDeviceMap[watchId] = setTimeout(generateDisconnectionAlert,
+                                               CONNECTION_TIMEOUT,
+                                               watchId);
+    } else {
+      createAlert({
+        timestamp: dataObj.timestamp,
+        watch_id: watchId,
+        alert_type: 'CONNECTION',
+        alert_text: 'Device has connected to DCH.'
+      });
+
+      // Reset the timeout
+      connectedDeviceMap[watchId] = setTimeout(generateDisconnectionAlert,
+                                               CONNECTION_TIMEOUT,
+                                               watchId);
+    }
+
+  } else {
     console.log('The format of the requested json was not correct');
   }
 });
 
+/**
+ * @desc - This function generates a 'disconnection' alert for a watch after
+ *         its connection timeout and removes it from the connectedDeviceMap
+ *
+ * @param uuid {string} - The UUID of the disconnected watch
+ */
+function generateDisconnectionAlert(uuid) {
+  var currentDate = new Date();
+  var dateInMilliseconds = currentDate.getTime();
+
+  var disconnectionAlert =
+    {
+      timestamp: dateInMilliseconds,
+      watch_id: uuid,
+      alert_type: 'CONNECTION',
+      alert_text: 'Device has disconnected from DCH.'
+    };
+
```

Comment:
Caught this bug last night. Need to put this object into the 'createAlert' function in order to broadcast it to DCGUI